addons/im_livechat/static/src/embed/core/livechat_service.js

```diff
@@ -73,6 +73,16 @@ export class LivechatService {
         this.rule = init.rule;
         this.initialized = true;
         this.initializedDeferred.resolve();
+
+        // Clear session if it is outdated.
+        const previousVersion = localStorage.getItem(this.ODOO_VERSION_KEY);
+        const currentVersion = init.odoo_version;
+        const visitorUid = this.visitorUid || false;
+        const userId = session.user_id || false;
+        if (previousVersion !== currentVersion || (this.sessionCookie && visitorUid !== userId)) {
+            this.leaveSession();
+        }
+        localStorage.setItem(this.ODOO_VERSION_KEY, init.odoo_version);
     }
 
     async _createSession({ persisted = false } = {}) {
```

Comment:
To do in follow-up PR:
It's not great to name state in past-tense, it makes things harder to read than it should.
Prefer `persist(ent): false`, `SESSION_STATE.NEW`, `SESSION_STATE.PERSISTENT`, etc.